### DISCUSSION TITLE
How Next.js API routes are handled?

### DISCUSSION THREAD

**Discussion by Rahul-Vignesh** [vercel/next.js#20600] [Help] [answered]
How are API routes in next.js handled by the server? Do they run on single node process ?

**Comment by lfades** [comment, marked as answer]
They are intended for serverless environments. e.g in Vercel each API route is a serverless function.

While using `next start` they work more like handlers where a single server takes care of everything.

**Reply by Rahul-Vignesh** [reply]
Is it clustered while running? I rewrote my express.js API to serverless architecture and have made few changes directly. But serverless wasn't the right fit for me after using it for sometime. So I want to know if I can rely on `next server` ?
